client/hosting/sites/index.tsx

```diff
@@ -25,6 +27,24 @@ export default function () {
 		navigation,
 		sitesDashboard,
 		makeLayout,
+		checkReferrerForSiteSelection,
 		clientRender
 	);
 }
+
+function checkReferrerForSiteSelection( context, next ) {
+	const { referrer } = document;
+	// Only evluate this on initialization. Note im not 100% sure if this init value is fully
+	// accurate for what we want, but on initial inspection it seems promising.
+	if ( ! context.init || ! referrer ) {
+		next();
+	}
+
+	const potentialSiteSlug = new URL( referrer ).hostname || '';
+	const referringSiteId = getSiteIdBySlug( context.store.getState(), potentialSiteSlug );
```

Comment:
Note - this does not work at this point where `context.init` is true. The site store does not seem to be setup fully as this always fails to return the siteId at this point.

If you comment out the early return for `context.init` above, you will see that this logic works when the store is fully initialized by going to the sites dashboard after loading the app elsewhere (like the reader or overview page in the dashboard).

I think we need to find a way to store the referrer and trigger checking to set it as the selected site once the store is initialized enough to find the siteId.